lci_strategic_plugin/include/lci_strategic_plugin/lci_strategic_plugin.h

```diff
@@ -0,0 +1,621 @@
+   * \param entry_dist distance to stop line
+   *
+   * \param current_speed current speed of vehicle
+   * 
+   * \param departure_speed speed to get into the intersection
+   *
+   * \return the time vehicle will stop with optimal decelarion
+   */
+  double calc_estimated_entry_time_left(double entry_dist, double current_speed, double departure_speed) const;
+
+  /**
+   * \brief Get boundary distances used to compare against current state in order to create trajectory smoothing parameters
+   *
+   * \param v0 starting_speed
+   * \param v1 departure_speed
+   * \param v_max speed_limit
+   * \param v_min minimum speed
+   * \param a_max max_comfort_acceleration limit
+   * \param a_min max_comfort_deceleration limit
+   * \return boundary distances used to generate trajectory smoothing segments
+   */
+  BoundaryDistances get_delta_x(double v0, double v1, double v_max, double v_min, double a_max, double a_min);
+  
+    /**
+   * \brief Get all possible trajectory smoothing parameters for each segments. Later this will be used to generate a single trajectory
+   * \param t current time in seconds
+   * \param v0 starting_speed
+   * \param v1 departure_speed
+   * \param v_max speed_limit
+   * \param v_min minimum speed
+   * \param a_max max_comfort_acceleration limit
+   * \param a_min max_comfort_deceleration limit
+   * \param x0 current_downtrack
+   * \param x_end traffic_light_down_track
+   * \param dx distance_remaining_to_traffic_light
+   * \param boundary_distances boundary_distances to compare against current state
+   * \return all possible trajectory smoothing parameters to later generate single trajectory
+   */
+  std::vector<TrajectoryParams> get_boundary_traj_params(double t, double v0, double v1, double v_max, double v_min, double a_max, double a_min, double x0, double x_end, double dx, BoundaryDistances boundary_distances);
+  
+  /**
+   * \brief Helper method to print TrajectoryParams
+   */
+  void print_params(TrajectoryParams params);
+
+  /**
+   * \brief Helper method to print TrajectoryParams
+   */
+  void print_boundary_distances(BoundaryDistances delta_xs);
+  
+  /**
+   * \brief Trajectory Smoothing Algorithm 8 cases.  TSMO UC2, UC3 algorithm equations
+   */
+  TrajectoryParams ts_case1(double t, double et, double v0, double v1, double v_max, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case2(double t, double et, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case3(double t, double et, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case4(double t, double et, double v0, double v1, double v_min, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case5(double t, double et, double v0, double a_max, double a_min, double x0, double x_end, double dx); //TODO Saeid: pls check dx, I included as it was missing
```

Comment:
Yes, dx must be added and is missing in the python script. Thanks for adding it.